plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -5591,6 +5595,33 @@ public void testGroupedExecution()
         };
     }
 
+    private Consumer<Plan> assertLocalRepartitionedExchangesCount(int expectedLocalExchangesCount)
+    {
+        return plan -> {
+            int actualLocalExchangesCount = searchFrom(plan.getRoot())
+                    .where(node -> {
+                        if (!(node instanceof ExchangeNode)) {
+                            return false;
+                        }
+
+                        ExchangeNode exchangeNode = (ExchangeNode) node;
+                        return exchangeNode.getScope() == ExchangeNode.Scope.LOCAL && exchangeNode.getType() == ExchangeNode.Type.REPARTITION;
+                    })
+                    .findAll()
+                    .size();
+            if (actualLocalExchangesCount != expectedLocalExchangesCount) {
+                Session session = getSession();
+                Metadata metadata = getDistributedQueryRunner().getCoordinator().getMetadata();
+                String formattedPlan = textLogicalPlan(plan.getRoot(), plan.getTypes(), metadata, StatsAndCosts.empty(), session, 0, false);
```

Comment:
Assertions printing a plan are tricky -- this is dead code most of the times (the code is correct), and thus is may not work. In fact, i did solve some problems like that in the past in other plan-related assertions, where upon plan mismatch  we would fail with "no transaction" when trying to print the plan.

Fortunately this isn't the case here yet, but ideally `assertLocalRepartitionedExchangesCount` should be tested separately to ensure the failure case is run. 

see for example https://github.com/trinodb/trino/blob/fbd1de01bba0c896e6e584f26d128452fdf35bc1/testing/trino-tests/src/test/java/io/trino/tests/AbstractQueryAssertionsTest.java#L104-L122

Reply:
I am OK ignoring this for now, but would be good to have a TODO comment